code/controllers/subsystem/changelog.dm

```diff
@@ -188,52 +189,64 @@ SUBSYSTEM_DEF(changelog)
 	// Load in the header
 	changelogHTML += changelog_header
 
-	// Make blocks for all the PRs
-	for(var/pr_number in prs_to_process)
-		// Initial declarations
-		var/pr_block = "" // HTML for the changelog section
-		var/author = "" // Author of the PR
-		var/merge_date = "" // Timestamp of when the PR was merged
+	// We put all these queries into a list so we can batch-execute them to avoid excess delays
+	// We index these based on PR numbers. MAKE SURE YOU USE STRING INDICIES IN THIS IF YOU EVER TWEAK IT -aa
+	var/list/datum/db_query/meta_queries = list()
+	var/list/datum/db_query/entry_queries = list()
 
-		// Now we gather the data from the DB
+	// Create some queries for each PR
+	for(var/pr_number in prs_to_process)
 		var/datum/db_query/pr_meta = SSdbcore.NewQuery(
 			"SELECT author, DATE_FORMAT(date_merged, '%Y-%m-%d at %T') AS date FROM changelog WHERE pr_number = :prnum LIMIT 1",
```

Comment:
This entire file is a complete mess. 

This whole thing can be accomplished with just one DB query that returns all the relevant records for PRs merged in the past 30 days. Group by PR, iterate over that and you've got a much cleaner solution than bombarding the DB with three layers of queries. Will run faster too.

Reply:
If you want to refactor this system to use less awful DB queries, you are more than welcome to.